pkg/sql/gcjob/gc_job.go

```diff
@@ -508,7 +508,7 @@ func shouldUseDelRange(
 ) bool {
 	// TODO(ajwerner): Adopt the DeleteRange protocol for tenant GC.
 	return details.Tenant == nil &&
-		s.Version.IsActive(ctx, clusterversion.TODODelete_V22_2UseDelRangeInGCJob) &&
+		s.Version.IsActive(ctx, clusterversion.V23_1_UseDelRangeInGCJob) &&
 		(storage.CanUseMVCCRangeTombstones(ctx, s) ||
```

Comment:
Shouldn't some of these predicates be somewhat different now? In mixed-version 22.2/23.1 clusters we should respect the `storage.mvcc.range_tombstones.enabled` setting until the `V23_1_MVCCRangeTombstonesUnconditionallyEnabled` version gate is enabled, at which point we should unconditionally use them. `CanUseMVCCRangeTombstones()` already encapsulates that logic.

The logic here will fail to respect `storage.mvcc.range_tombstones.enabled` in mixed-version 22.2/23.1 clusters.

Reply:
You're correct, it's done.